replica.go

```diff
@@ -66,6 +66,9 @@ func RemoveAllTagsOfReplica() ReplicaOption {
 
 // WithDemandOfReplica will add or overwrite the amount of a target Resource the replica demands
 func WithDemandOfReplica(targetResource Resource, demandAmount int64) ReplicaOption {
+	if demandAmount < 0 {
+		panic("resourceAmount cannot be negative")
+	}
```

Comment:
Lots of the same checks, could potential make the `demandAmount` a custom type and implement a `.validate()` method that we can call, but seems overkill for a three line check used 3-5 times, the cost to first cast it and change it everywhere for the custom type to be accepted is not appealing to my 4AM self.